lms/djangoapps/courseware/views/views.py

```diff
@@ -2135,6 +2160,8 @@ def get_financial_aid_courses(user):
                 }
             )
 
+    if course_id is not None and len(financial_aid_courses) != 1:
+        return []
```

Comment:
what's the thought process behind this check?

Reply:
If a course_id (say, an invalid course id) is given, the loop above may get more than one courses. This condition can be improved so let me update it.